framework/encode/capture_manager.cpp

```diff
@@ -1010,21 +1057,63 @@ std::string CommonCaptureManager::CreateTrimDrawCallsFilename(const std::string&
     return util::filepath::InsertFilenamePostfix(base_filename, range_string);
 }
 
+std::string CommonCaptureManager::CreateAssetFile()
+{
+    std::string asset_file_name = CreateAssetFilename(base_filename_);
+
+    if (timestamp_filename_)
+    {
+        asset_file_name = util::filepath::GenerateTimestampedFilename(asset_file_name);
+    }
+
+    asset_file_stream_ = std::make_unique<util::FileOutputStream>(asset_file_name, kFileStreamBufferSize);
+    if (asset_file_stream_->IsValid())
+    {
+        WriteFileHeader(asset_file_stream_.get());
+    }
+    else
+    {
+        asset_file_stream_ = nullptr;
+    }
+
+    return asset_file_name;
+}
+
+std::string CommonCaptureManager::CreateAssetFilename(const std::string& base_filename) const
+{
+    std::string asset_filename = base_filename;
+
+    size_t dot_pos = base_filename.rfind('.');
```

Comment:
Please see if the functions `GetFilenameStem` and `GetFilenameExtension` in `gfxrecon::util::filepath` can be used for parsing out the extension.